src/cc_mm.c

```diff
@@ -34,10 +34,8 @@ _cc_alloc(size_t size, const char *name, int line)
 {
     void *p;
 
-    ASSERT(size != 0);
-
     p = malloc(size);
-    if (p == NULL) {
+    if (p == NULL && size != 0) {
```

Comment:
From ccommon's perspective, I think we should remove such ambiguity (e.g. implementation-defined behavior), and always return a `NULL` pointer (since the address is never meant to be accessed by the caller). Hence having the check and return value set at the top of the function.